### DISCUSSION TITLE
Config plugins are only required for React Native Firebase modules - but which modules are these?

### DISCUSSION THREAD

**Discussion by Nantris** [invertase/react-native-firebase#6510] [Q&A] [answered]
From this page: https://rnfirebase.io/#expo

> **Config plugins are only required for React Native Firebase modules, which require custom native installation steps** - e.g. modifying the Xcode project, Podfile, build.gradle, AndroidManifest.xml etc. Packages without native steps required will work out of the box.

but 

> **Not all React Native Firebase packages have Expo config plugins provided yet.** You may see if a module is supported by checking if it contains the app.plugin.js file in its package directory.

But how do we know if a package not having an `app.plugin.js` means it that doesn't need a plugin, as opposed to the plugin just not existing yet?

Could we get a comprehensive list of modules which require plugins to work?

And if plugins are required for `auth`, `storage`, and/or `database`, are those works in progress, or only hopes at this point? If we need all of those in Expo Managed workflow, should we hold off on implementing `react-native-firebase`?

Thanks very much for the great work and for reading.

**Comment by mikehardy** [comment, marked as answer]
> Could we get a comprehensive list of modules which require plugins to work?

Check the documentation on rnfirebase.io for install steps on ios and android, if the install steps indicate that you need to modify native files and there is no app.plugin.js, then you know it needs one and does not have one.

> Could we get a comprehensive list of modules which require plugins to work?

Expo support is 100% provided by the community of Expo users. Please post a PR to our docs after investigating the above if this is something you need

> And if plugins are required for auth, storage, and/or database, are those works in progress, or only hopes at this point? If we need all of those in Expo Managed workflow, should we hold off on implementing react-native-firebase?

I *think* it is down to just needing a plugin for auth package *if you need phone auth*, but I'm not sure. If it's something you need, don't hold off, just implement the plugin (or finish this one that's going stale because author has some sort of github account issue? #5787 / #6472 / #6167) and you're set

**Reply by Nantris** [reply]
Thanks very much for the reply @mikehardy!

> Check the documentation on rnfirebase.io for install steps on ios and android

I think I understand, but I'm not crystal clear on that. You mean basically if a sub-component of the library has a section like this specified in the docs, it requires a plugin, right? https://rnfirebase.io/perf/usage#add-the-performance-monitoring-plugin

**Reply by mikehardy** [reply]
Exactly that!
